src/Boids.hs

```diff
@@ -31,23 +30,22 @@ renderBoid boid = translate x y $ rotate (-heading) $ polygon ps
           (x, y) = boidPosition boid
 
 nextBoid :: Float -> Boid -> Boid
-nextBoid deltaTime boid = boid {
-                            boidPosition = (x + deltaTime * cos heading * 100.0, y + deltaTime * sin heading * 100.0)
-                          }
+nextBoid deltaTime boid = boid { boidPosition = (pos x cos, pos y sin) }
     where (x, y) = boidPosition boid
           heading = boidHeading boid
-
-randomBoid :: IO Boid
-randomBoid = do x <- randomRIO (-100.0, 100.0)
-                y <- randomRIO (-100.0, 100.0)
-                heading <- randomRIO (0.0, 2 * pi)
-                return $ Boid { boidPosition = (x, y)
-                              , boidHeading = heading
-                              , boidSteer = 0.0
-                              }
-
-initialState :: IO World
-initialState = replicateM 100 randomBoid
+          pos axis f = axis + deltaTime * (f heading) * 100.0
+
+randomBoid :: [Float] -> Boid
+randomBoid salts = Boid { boidPosition = (getPos s1, getPos s2)
+                        , boidHeading = saltedRange s3 (0, 2 * pi)
+                        , boidSteer = 0.0
+                        }
+  where [s1, s2, s3] = salts
+        getPos s = saltedRange s (-100, 100)
+
+initialState :: Int -> [Float] -> World
+initialState boidsCount salts = [ randomBoid (dosedLists 3 salts !! i)
```

Comment:
@Minoru Thanks! I totally agree, I just don't know what I was thinking about when was making this strange solution instead of just simple and neat recursive function.